shell/platform/android/io/flutter/plugin/common/MethodCall.java

```diff
@@ -0,0 +1,29 @@
+package io.flutter.plugin.common;
+
+import java.util.Objects;
+
+/**
+ * Command object representing a method call on a {@link FlutterChannel}.
+ */
+public final class MethodCall {
+    /**
+     * The name of the called method.
+     */
+    public final String method;
+
+    /**
+     * Arguments for the call.
+     */
+    public final Object arguments;
```

Comment:
Sorry, the "price" is the extra codec and channel concepts. I'd be happy to give it a go, but I'm not sure I understand what we expect to gain by having two channel types. Is it mainly to provide an API with more precise types? Or to reserve the ability to define alternative communication protocols similar to method calls and event streams without having to add methods to a single channel and codec concept?

We had a discussion at the office about whether to have FlutterChannel<T> with T being the type of message/method call arguments/reply envelope contents. Then we'd have FlutterChannel<String> superseding the existing string protocol methods, and FlutterChannel<ByteBuffer> for binary. But we ended up going with a non-generic FlutterChannel with Object instead of T, because we expect most people to need the flexibility of the json or standard codecs, where all the Xxx<Object> on channels and call-backs are just clutter, adding no information.